projects/opendr_ws_2/src/opendr_perception/opendr_perception/object_detection_2d_detr_node.py

```diff
@@ -47,64 +52,168 @@ def __init__(self, input_rgb_image_topic="image_raw", output_rgb_image_topic="/o
 
-        # Run object detection
-        boxes = self.object_detector.infer(image)
+        # Run detection estimation
+        boxes = self.detr_learner.infer(image)
 
         # Get an OpenCV image back
-        image = image.opencv()
+        image = np.float32(image.opencv())
 
-        # Publish detections in ROS message
-        ros_boxes = self.bridge.to_ros_bounding_box_list(boxes)  # Convert to ROS bounding_box_list
-        if self.object_publisher is not None:
-            self.object_publisher.publish(ros_boxes)
+        #  Annotate image and publish results:
+        if self.detection_publisher is not None:
+            ros_detection = self.bridge.to_ros_bounding_box_list(boxes)
+            self.detection_publisher.publish(ros_detection)
+            # We get can the data back using self.bridge.from_ros_bounding_box_list(ros_detection)
+            # e.g., opendr_detection = self.bridge.from_ros_bounding_box_list(ros_detection)
 
         if self.image_publisher is not None:
-            # Annotate image with object detection boxes
-            image = draw(image, boxes)
-            # Convert the annotated OpenDR image to ROS2 image message using bridge and publish it
-            self.image_publisher.publish(self.bridge.to_ros_image(Image(image), encoding='bgr8'))
+            image = draw_bounding_boxes(image, boxes, class_names=self.class_names)
+            message = self.bridge.to_ros_image(Image(image), encoding="bgr8")
+            self.image_publisher.publish(message)
 
 
 def main(args=None):
     rclpy.init(args=args)
 
     parser = argparse.ArgumentParser()
-    parser.add_argument("-i", "--input_rgb_image_topic", help="Topic name for input rgb image",
-                        type=str, default="image_raw")
-    parser.add_argument("-o", "--output_rgb_image_topic", help="Topic name for output annotated rgb image",
-                        type=str, default="/opendr/image_objects_annotated")
-    parser.add_argument("-d", "--detections_topic", help="Topic name for detection messages",
-                        type=str, default="/opendr/objects")
-    parser.add_argument("--device", help="Device to use, either \"cpu\" or \"cuda\", defaults to \"cuda\"",
-                        type=str, default="cuda", choices=["cuda", "cpu"])
+    parser.add_argument("-i", "--input_rgb_image_topic", help="Topic name for input rgb image", type=str, default="image_raw")
+    parser.add_argument(
+        "-o",
+        "--output_rgb_image_topic",
+        help="Topic name for output annotated rgb image",
+        type=str,
+        default="/opendr/image_objects_annotated",
+    )
+    parser.add_argument(
+        "-d", "--detections_topic", help="Topic name for detection messages", type=str, default="/opendr/objects"
+    )
+    parser.add_argument(
+        "--device",
+        help='Device to use, either "cpu" or "cuda", defaults to "cuda"',
+        type=str,
+        default="cuda",
+        choices=["cuda", "cpu"],
+    )
```

Comment:
Could you make the formatting a bit more consistent, maybe similar to line 199 above for the `input_rgb_image_topic`, to improve readability?

Reply:
Unfortunately this would result in a violation of the max line length. I used a formatting tool to format the code (Black) so it should be quite consistent actually, but if you have other preferences then I can modify it.

Reply:
Alright, it's a minor thing so we can just keep it this way! If you are done with the changes, feel free to re-request a review.